### PR TITLE
Use Alpine instead of Ubuntu Trusty

### PR DESCRIPTION
**Pull request by nelljerram**
Alpine as a base should be more secure.  Also this gives us an image
size of 105MB, as against 292MB with Trusty.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,28 +11,31 @@
...
-#ADD /image/buildconfig /build/buildconfig
-#ADD /image/my_init /build/my_init
-#ADD /image/install.sh /build/install.sh
-#RUN /build/install.sh
-#ADD /image/system_services.sh /build/system_services.sh
-#RUN	/build/system_services.sh
-#ADD /image/cleanup.sh /build/cleanup.sh
-#RUN	/build/cleanup.sh
+RUN echo "ipv6" >> /etc/modules
+RUN echo "http://dl-1.alpinelinux.org/alpine/edge/main" >> /etc/apk/repositories; \
```
Comment (fasaxc) [inline Dockerfile:22]: As discussed offline, if we're going to use edge, I think we should use an edge base image, which should remove the need to add these repos.

**Comment by nelljerram** [comment]
Not going to proceed with this now, because BIRD packages are not yet in a proper Alpine release.  But will leave this PR open in the expectation that they will be in a proper release soon.

**Comment by djlwilder** [comment]
@neiljerram what is the issue you saw with bird in alpine?  I am asking as I need to build for ppc64le and need to select a new base image (I cant use 14:04).  If your planning a switch to alpine I would like to follow your lead.

**Comment by nelljerram** [comment]
@djlwilder As above, it was just that BIRD is not yet available in any numbered Alpine release.

**Comment by mkumatag** [comment]
Now I see bird packages are there in the edge [release](https://pkgs.alpinelinux.org/packages?name=bird&branch=edge), so are we waiting for Alpine 3.8 release(happen once in a 6months) or going with edge release which already has bird package.?

**Comment by nelljerram** [comment]
@mkumatag Yes, we're waiting - and TBH in a pretty non-proactive way, because I think we're seeing this PR as just a nice-to-have, rather than something that is actually important.

Now, if you have more reason to be getting on with this, it would be worth looking - for example - at whether there is a way of asking the Alpine people to move BIRD into the next numbered release.

**Comment by mkumatag** [comment]
@neiljerram Every release of Alpine is a snapshot of Edge release as per [Alpine Release Page](https://wiki.alpinelinux.org/wiki/Alpine_Linux:Releases), so hopefully in next release(v3.8 in May) will have bird package.

**Comment by nelljerram** [comment]
@mkumatag Are you sure?  The history of BIRD in Alpine seems to go back a long time: https://git.alpinelinux.org/cgit/aports/log/testing/bird/APKBUILD.  So if what you say is true, I would expect BIRD to already be in Alpine numbered releases.

**Comment by mkumatag** [comment]
I just had quick chat with the alpine community over IRC and got to know the package should be in a stable release to get it included in the next numbered release. As Bird package is in testing repo, so we have to talk to maintainer and ask him to promote this package to stable(main) repo. 

Got the information about the maintainer of the package, will talk to him and let you know the further updates, stay tuned.

**Comment by mkumatag** [comment]
@neiljerram I see Bird package shipped with the alpine distro is referring to http://bird.network.cz/ and not the one present in calicoproject, so are we okay with that?

IMO if we are having our own fork of bird repo then we should be using that instead of http://bird.network.cz/ or are there any specific reason behind choosing the bird package from distro than calico one?

**Comment by nelljerram** [comment]
@mkumatag Thanks for writing to the Alpine community!

Yes, for route reflector purposes, upstream BIRD is absolutely fine.  Our BIRD fork is only needed on compute nodes when IP-in-IP is being used because compute nodes are not directly connected.

**Comment by tomdee** [comment]
@neiljerram Do you intend to do further work on this PR or can it be closed?

**Comment by nelljerram** [comment]
@tomdee Yes, this can be closed now.  I have a task planned to retire this repo completely, which would also have cleaned up this PR; but there's no particular reason to wait until I get to that.

**Pull request closed without merging**